src/ORM/Rule/IsUnique.php

```diff
@@ -63,6 +64,15 @@ public function __invoke(EntityInterface $entity, array $options)
             }
         }
 
+        if ($checkNull) {
+            foreach ($conditions as $key => $value) {
```

Comment:
@markstory So add another parameter to the [alias](https://github.com/cakephp/cakephp/blob/master/src/ORM/Table.php#L354) method? (Currently the only parameter is `$alias`)